mesonbuild/wrap/wrap.py

```diff
@@ -383,18 +384,44 @@ def download(self, what: str, ofname: str) -> None:
         os.rename(tmpfile, ofname)
 
     def get_file_internal(self, what: str) -> str:
-        filename = self.wrap.get(what + '_filename')
-        cache_path = os.path.join(self.cachedir, filename)
+        if all([x in self.wrap.values for x in [what + '_url', what + '_path']]):
+            raise WrapException('Both {}_url and {}_path were specified'.format(what, what))
 
-        if os.path.exists(cache_path):
-            self.check_hash(what, cache_path)
-            mlog.log('Using', mlog.bold(self.packagename), what, 'from cache.')
+        if what + '_url' in self.wrap.values:
+            filename = self.wrap.get(what + '_filename')
+            cache_path = os.path.join(self.cachedir, filename)
+
+            if os.path.exists(cache_path):
+                self.check_hash(what, cache_path)
+                mlog.log('Using', mlog.bold(self.packagename), what, 'from cache.')
+                return cache_path
+
+            if not os.path.isdir(self.cachedir):
+                os.mkdir(self.cachedir)
+            self.download(what, cache_path)
             return cache_path
+        elif what + '_path' in self.wrap.values:
+            from ..interpreterbase import FeatureNew
+            FeatureNew('Local wrap patch files via {}_path'.format(what), '0.54.0').use('')
```

Comment:
You should pass down current interpreter's subproject, it's not always the main project that use that new feature.

Reply:
Fixed.